extensions/data-transfer/portability-data-transfer-google/src/test/java/org/dataportabilityproject/datatransfer/google/mail/GoogleMailExporterTest.java

```diff
@@ -16,6 +16,157 @@
 
 package org.dataportabilityproject.datatransfer.google.mail;
 
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+import com.google.api.services.gmail.Gmail;
+import com.google.api.services.gmail.Gmail.Users;
+import com.google.api.services.gmail.Gmail.Users.Messages;
+import com.google.api.services.gmail.Gmail.Users.Messages.Get;
+import com.google.api.services.gmail.model.ListMessagesResponse;
+import com.google.api.services.gmail.model.Message;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.dataportabilityproject.datatransfer.google.common.GoogleCredentialFactory;
+import org.dataportabilityproject.spi.transfer.provider.ExportResult;
+import org.dataportabilityproject.spi.transfer.types.ContinuationData;
+import org.dataportabilityproject.spi.transfer.types.ExportInformation;
+import org.dataportabilityproject.spi.transfer.types.PaginationData;
+import org.dataportabilityproject.spi.transfer.types.StringPaginationToken;
+import org.dataportabilityproject.types.transfer.models.mail.MailContainerResource;
+import org.dataportabilityproject.types.transfer.models.mail.MailMessageModel;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InOrder;
+import org.mockito.Matchers;
+import org.mockito.Mockito;
+
 public class GoogleMailExporterTest {
 
+  private static final UUID JOB_ID = UUID.randomUUID();
+
+  private static final String NEXT_TOKEN = "next_token";
+
+  private static final String MESSAGE_ID = "messageId";
+  private static final String MESSAGE_RAW = "message contents";
+  private static final List<String> MESSAGE_LABELS = ImmutableList.of("label1", "label2");
+  private static final Message INITIAL_MESSAGE = new Message().setId(MESSAGE_ID);
+  private static final Message FULL_MESSAGE = new Message().setId(MESSAGE_ID).setRaw(MESSAGE_RAW)
+      .setLabelIds(MESSAGE_LABELS);
+
+  private Users users;
```

Comment:
if you add @Mock annotation to the ones that are mocks, you can omit the lines in setup